### PR TITLE
Fix purl lookups

### PR DESCRIPTION
**Pull request by TG1999**

### CODE REVIEW

**Comment by tdruez** [comment]
- Tests are failing
- Missing changelog entry

**Comment by TG1999** [comment]
@tdruez got this 
```
src/packageurl/contrib/django/models.py:27: in <module>
    from django.core.exceptions import ValidationError
E   ModuleNotFoundError: No module named 'django'
``` 
I have not modified line 27, I can't understand why this is failing, can you please help me?

**Comment by TG1999** [comment]
I tried to put `django` in setup.cfg,
```
[options.extras_require]
test =
    pytest
    isort
    black
    django
```
but then I got this 
```
src/packageurl/contrib/django/models.py:89: in <module>
    class PackageURLMixin(models.Model):
lib/python3.8/site-packages/django/db/models/base.py:127: in __new__
    app_config = apps.get_containing_app_config(module)
lib/python3.8/site-packages/django/apps/registry.py:260: in get_containing_app_config
    self.check_apps_ready()
lib/python3.8/site-packages/django/apps/registry.py:137: in check_apps_ready
    settings.INSTALLED_APPS
lib/python3.8/site-packages/django/conf/__init__.py:92: in __getattr__
    self._setup(name)
lib/python3.8/site-packages/django/conf/__init__.py:72: in _setup
    raise ImproperlyConfigured(
E   django.core.exceptions.ImproperlyConfigured: Requested setting INSTALLED_APPS, but settings are not configured. You must either define the environment variable DJANGO_SETTINGS_MODULE or call settings.configure() before accessing settings.
```
----

>Especially since all intermediate commits can be merged as a single one when merging the PR.

I try to keep git history clean by merging commits which have small incremental changes or commits in which I am experimenting with something, but yeah this makes sense.

File: src/packageurl/contrib/django/models.py [hunk 1]
```diff
@@ -24,53 +24,33 @@
...
-    except ValueError:
-        return  # Not a valid PackageURL
-
-    package_url_dict = package_url.to_dict(encode=True)
-    return without_empty_values(package_url_dict)
+warnings.warn(
+    "`purl_to_lookups`, `without_empty_values` is moved from packageurl.contrib.django.models"
+    " to packageurl.contrib.django.utils",
+    DeprecationWarning,
+)
```
Comment (tdruez) [inline src/packageurl/contrib/django/models.py:40]: You need to ensure that current imports of without_empty_values and purl_to_lookups from the models module are still working for backward compatibility.
Also, you should only warn when importing from `models`, you do not want to warn when loading this module.

File: src/packageurl/contrib/django/models.py [hunk 2]
```diff
@@ -32,13 +32,17 @@
...
+from packageurl.contrib.django.utils import without_empty_values
 
 warnings.warn(
     "`purl_to_lookups`, `without_empty_values` is moved from packageurl.contrib.django.models"
     " to packageurl.contrib.django.utils",
     DeprecationWarning,
 )
 
+purl_to_lookups = purl_to_lookups
+without_empty_values = without_empty_values
```
Comment (tdruez) [inline src/packageurl/contrib/django/models.py:44]: Those statements are useless.

File: src/packageurl/contrib/django/utils.py [hunk 3]
```diff
@@ -33,6 +35,11 @@ def purl_to_lookups(purl, encode=True):
     Return a lookups dict built from the provided `purl` string.
     Those lookups can be used as QuerySet filters.
     """
+    warnings.warn(
+        "`purl_to_lookups` is moved from packageurl.contrib.django.models"
+        " to packageurl.contrib.django.utils",
+        DeprecationWarning,
+    )
```
Comment (tdruez) [inline src/packageurl/contrib/django/utils.py:42]: We should not warn here.

Warning should only be raised when the following import are made:
- `from packageurl.contrib.django.models import purl_to_lookups`
- `from packageurl.contrib.django.models import without_empty_values`

Why warning one that is properly importing from utils?

Comment (TG1999) [reply src/packageurl/contrib/django/utils.py]: Yeah I messed up that commit, made some changes in a new commit

File: src/packageurl/contrib/django/models.py (hunk 1 above)
Comment (TG1999) [reply src/packageurl/contrib/django/models.py]: Thanks! Made the requested changes

File: src/packageurl/contrib/django/models.py (hunk 2 above)
Comment (TG1999) [reply src/packageurl/contrib/django/models.py]: Yeah I messed up that commit, made some changes in a new commit

File: src/packageurl/contrib/django/utils.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# Visit https://github.com/package-url/packageurl-python for support and
+# download.
+
+
+import warnings
```
Comment (tdruez) [inline src/packageurl/contrib/django/utils.py:28]: Left over.

**Review by tdruez — DISMISSED** [review]

**Review by pombredanne — DISMISSED** [review]
Thanks. LGTM! :bow:

**Review by pombredanne — APPROVED** [review]
:+1:

**Pull request merged**